### PR TITLE
INN-1348 Throw an actionable error when we detect mixed async logic

### PR DESCRIPTION
**Pull request by jpwilliams**
## Related

- #187 
- INN-1348

### CODE REVIEW

File: src/components/InngestFunction.ts [hunk 1]
```diff
@@ -485,6 +499,30 @@ export class InngestFunction<
...
+         * promises to return no further actions, so we should only throw if
+         * this state is reached and there are no other pending steps.
+         */
+        const hasOpsPending = Object.values(state.allFoundOps).some((op) => {
+          return op.fulfilled === false;
+        });
+
+        if (!hasOpsPending) {
+          throw new Error(
+            functionStoppedRunningErr(ErrCode.ASYNC_DETECTED_AFTER_MEMOIZATION)
```
Comment (jpwilliams) [inline src/components/InngestFunction.ts:523]: Should throwing any of these errors disable retries? e.g. return `4xx`. I think yes.

Comment (darwin67) [reply src/components/InngestFunction.ts:523]: I agree retries should not attempt since there's no point until user update their code

**Review by darwin67 — APPROVED** [review]

**Comment by darwin67** [comment]
Is there a follow up task to communicate this change to users when releasing?
Best people are aware and know what changed in the new release so people's code don't blow up all of a sudden, and we don't suddenly get a spike of support requests.

**Review by tonyhb — APPROVED** [review]

**Comment by tonyhb** [comment]
> Is there a follow up task to communicate this change to users when releasing? Best people are aware and know what changed in the new release so people's code don't blow up all of a sudden, and we don't suddenly get a spike of support requests.

I dont think we need this, as existing code is already broken and wouldn't have been functioning.  Now, users will at least start seeing errors.

When we enable this, via an unsafe flag, we should probably let users know.

**Pull request merged**